sdk/ai/ai-projects/README.md

```diff
@@ -0,0 +1,711 @@
+) {
+  // Wait for a second
+  await new Promise((resolve) => setTimeout(resolve, 1000));
+  run = await client.agents.getRun(thread.id, run.id);
+}
+```
+
+To have the SDK poll on your behalf, use the `createThreadAndRun` method.
+
+Here is an example:
+
+```javascript
+const run = await client.agents.createThreadAndRun(thread.id, agent.id);
+```
+
+With streaming, polling also need not be considered.
+
+Here is an example:
+
+```javascript
+const streamEventMessages = await client.agents.createRun(thread.id, agent.id).stream();
+```
+
+Event handling can be done as follows:
+
+```javascript
+for await (const eventMessage of streamEventMessages) {
+switch (eventMessage.event) {
+  case RunStreamEvent.ThreadRunCreated:
+    console.log(`ThreadRun status: ${(eventMessage.data as ThreadRunOutput).status}`)
+    break;
+  case MessageStreamEvent.ThreadMessageDelta:
+    {
+      const messageDelta = eventMessage.data as MessageDeltaChunk;
+      messageDelta.delta.content.forEach((contentPart) => {
+        if (contentPart.type === "text") {
+          const textContent = contentPart as MessageDeltaTextContent
+          const textValue = textContent.text?.value || "No text"
+          console.log(`Text delta received:: ${textValue}`)
+        }
+      });
+    }
+    break;
+
+  case RunStreamEvent.ThreadRunCompleted:
+    console.log("Thread Run Completed");
+    break;
+  case ErrorEvent.Error:
+    console.log(`An error occurred. Data ${eventMessage.data}`);
+    break;
+  case DoneEvent.Done:
+    console.log("Stream completed.");
+    break;
+  }
+}
+```
+
+#### Retrieve Message
+
+To retrieve messages from agents, use the following example:
+
+```javascript
+const messages = await client.agents.listMessages(thread.id);
+
+// The messages are following in the reverse order,
+// we will iterate them and output only text contents.
+for (const dataPoint of messages.data.reverse()) {
+    const lastMessageContent: MessageContentOutput = dataPoint.content[dataPoint.content.length - 1];
+    console.log( lastMessageContent);
+    if (isOutputOfType<MessageTextContentOutput>(lastMessageContent, "text")) {
+      console.log(`${dataPoint.role}: ${(lastMessageContent as MessageTextContentOutput).text.value}`);
+    }
+  }
+```
+
+### Retrieve File
+
+Files uploaded by Agents cannot be retrieved back. If your use case need to access the file content uploaded by the Agents, you are advised to keep an additional copy accessible by your application. However, files generated by Agents are retrievable by `getFileContent`.
+
+Here is an example retrieving file ids from messages:
+
+```javascript
+const messages = await client.agents.listMessages(thread.id);
+const imageFile = (messages.data[0].content[0] as MessageImageFileContentOutput).imageFile;
+const imageFileName = (await client.agents.getFile(imageFile.fileId)).filename;
+
+const fileContent = await (await client.agents.getFileContent(imageFile.fileId).asNodeStream()).body;
+if (fileContent) {
```

Comment:
nice to have: we could pipe the stream body directly to a fs write stream. for example https://github.com/Azure/azure-sdk-for-js/blob/cbd2b87f97f5c463cc3e647b4cf35cd907a45d69/sdk/maps/maps-render-rest/samples-dev/getMapStaticImage.ts#L48

Reply:
Will address in separate PR

Reply:
Will add in samples and readme for next release - thanks!